lib/common.ts

```diff
@@ -236,8 +242,45 @@ export async function hashDataset(state: C14nState, quads: Iterable<rdf.Quad>, s
  * @param nquads 
  * @returns parsed dataset
  */
-export function parseNquads(nquads: string): Quads {
+export function parseNquads(nquads: string): InputQuads {
     const parser = new n3.Parser({ blankNodePrefix: '' });
     const quads: rdf.Quad[] = parser.parse(nquads);
-    return new Set<rdf.Quad>(quads);
+    return new n3.Store(quads);
+}
+
+
+/** TypeScript version of the TermSet class found in @rdfjs/term-set  */
+
+export class BnodeSet {
+    private index: Map<string, rdf.BlankNode>;
+
+    constructor() {
+        this.index = new Map();
+    }
+
+    get size() {
+        return this.index.size;
+    }
+
+    add(term: rdf.BlankNode) {
+        const key = term.value;
+
+        if (!this.index.has(key)) {
+            this.index.set(key, term);
+        }
+
+        return this;
+    }
+
+    values() {
+        return new Set(this.index.values());
+    }
+
+    keys() {
+        return this.values();
+    }
+
+    [Symbol.iterator]() {
+        return this.values()[Symbol.iterator]();
```

Comment:
Avoids unnecessary generation of an intermediary set
 
```suggestion
        return this.index.values();
```

Reply:
Yeah... I was wondering why it was so in the TermSet instance, but I though "these guys probably know something I don't..." :-)